package/gluon-wan-dnsmasq/files/etc/init.d/gluon-wan-dnsmasq

```diff
@@ -2,28 +2,18 @@
 
 START=60
 
-SERVICE_NAME=gluon-wan-dnsmasq
-SERVICE_USE_PID=1
-SERVICE_PID_FILE=/var/run/gluon-wan-dnsmasq.pid
-
-
-PORT=54
-PACKET_MARK=1
+USE_PROCD=1
 
 RESOLV_CONF_DIR=/var/gluon/wan-dnsmasq
 RESOLV_CONF=$RESOLV_CONF_DIR/resolv.conf
 
-
-start() {
+start_service() {
 	mkdir -p $RESOLV_CONF_DIR
 	/lib/gluon/wan-dnsmasq/update.lua
```

Comment:
I just noticed that there may be a race condition between the init script and the hotplug.d call of `/lib/gluon/wan-dnsmasq/update.lua` (unrelated to this PR, but we might as well fix it now).

I think it should be sufficient to replace the `update.lua` call here with a `touch "$RESOLV_CONF"`: If the WAN interface is not up at this point yet, `update.lua` would just have an empty `resolv.conf` here, and if it is, a proper `resolv.conf` has been (or is currently being) created by the hotplug script.